tests/Rule/EnsurePhpReferenceSyntaxTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of DOCtor-RST.
+ *
+ * (c) Oskar Stark <oskarstark@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Tests\Rule;
+
+use App\Rule\EnsurePhpReferenceSyntax;
+use App\Tests\RstSample;
+use App\Value\NullViolation;
+use App\Value\Violation;
+use App\Value\ViolationInterface;
+
+final class EnsurePhpReferenceSyntaxTest extends \App\Tests\UnitTestCase
+{
+    /**
+     * @test
+     *
+     * @dataProvider checkProvider
+     */
+    public function check(ViolationInterface $expected, RstSample $sample): void
+    {
+        self::assertEquals(
+            $expected,
+            (new EnsurePhpReferenceSyntax())->check($sample->lines(), $sample->lineNumber(), 'filename'),
+        );
+    }
+
+    /**
+     * @return \Generator<array{0: ViolationInterface, 1: RstSample}>
+     */
+    public static function checkProvider(): \Generator
+    {
+        yield [
+            NullViolation::create(),
+            new RstSample('temp'),
+        ];
```

Comment:
Maybe we can introduce an AbstractTestCase? to remove some code duplication?

(In a follow up PR ofc)